app/services/gitlab/fetch_branches_service.rb

```diff
@@ -0,0 +1,33 @@
+module Gitlab
+  class FetchBranchesService
+    delegate :error_message, to: :api, allow_nil: true
+
+    def initialize(project)
+      @project = project
+    end
+
+    def fetch
```

Comment:
not sure if I misread the code, but where are you calling this method and branches method?

Reply:
At this moment, nowhere, the whole purpose of this pr is to set the backend for the front end to work on it latter, Dieggo told me it'd be better to let the controller be developed by the front end.